src/client/dfs/cont.c

```diff
@@ -15,6 +16,67 @@
 
 #include "dfs_internal.h"
 
+static int
+suggest_dfs_cs(daos_handle_t poh, daos_prop_t *prop, uint64_t rf, daos_oclass_id_t oc_id,
+	       daos_size_t *cs)
+{
+	struct daos_oclass_attr *oc_attr;
+	struct daos_prop_entry  *dpe;
+	uint64_t                 ec_cell_size;
+	uint64_t                 val;
+	uint32_t                 nr_grps;
+	int                      rc;
+
+	/** No EC above RF 2, use default CS */
+	if (rf > 2) {
+		*cs = DFS_DEFAULT_CHUNK_SIZE;
+		return 0;
+	}
+
+	if (oc_id == 0) {
+		daos_obj_id_t oid       = {.hi = 0, .lo = 0};
+		uint32_t      pa_domain = daos_cont_prop2redunlvl(prop);
+
+		/** generate the oclass that would be used for file  */
+		rc = daos_obj_generate_oid_by_rf(poh, rf, &oid, DAOS_OT_ARRAY_BYTE, OC_UNKNOWN, 0,
+						 0, pa_domain);
+		if (rc) {
+			D_ERROR("daos_obj_generate_oid_by_rf() Failed: " DF_RC "\n", DP_RC(rc));
+			return daos_der2errno(rc);
+		}
+
+		oc_attr = daos_oclass_attr_find(oid, &nr_grps);
+		if (oc_attr == NULL)
+			return EINVAL;
+	} else {
+		oc_attr = daos_oclass_id2attr(oc_id, &nr_grps);
+		if (oc_attr == NULL)
+			return EINVAL;
+	}
+
+	/** for Replication (including non-redundant), return the default chunk size - 1 MiB */
+	if (oc_attr->ca_resil == DAOS_RES_REPL) {
+		*cs = DFS_DEFAULT_CHUNK_SIZE;
+		return 0;
+	}
+
+	/** query the EC cell size */
+	dpe = daos_prop_entry_get(prop, DAOS_PROP_CO_EC_CELL_SZ);
+	if (dpe)
+		ec_cell_size = dpe->dpe_val;
+	else
+		ec_cell_size = DAOS_EC_CELL_DEF;
+
+	/** set the DFS chunk size to the 2 x the EC cell size x the number of data cells */
+	val = oc_attr->u.ec.e_k * ec_cell_size * 2;
+	if (val <= DFS_DEFAULT_CHUNK_SIZE)
+		*cs = DFS_DEFAULT_CHUNK_SIZE;
```

Comment:
just confirm any reason why chunk size must >= 1MB?
For example if user define cell size as 100KB, for EC4P2, the "oc_attr->u.ec.e_k * ec_cell_size * 2" is 800K, why must set the chunk size to 1MB that is not full-stripe aligned?

Reply:
no reason. ill fix this.